g2o/types/slam2d_addons/edge_line2d.h

```diff
@@ -79,7 +79,7 @@ namespace g2o {
       }
 
 
-      virtual number_t initialEstimatePossible(const OptimizableGraph::VertexSet& , OptimizableGraph::Vertex* ) { return 0.;}
```

Comment:
same as above

Reply:
Thanks - these casts weren't done in the original files, which is why I left them as-is.

Reply:
Worth doing the casts as a separate merge?

Reply:
Whatever is easiest. Since these are all in the general cleanup kind of bin (and aren't large), I don't see why you'd pay the overhead of splitting it up into two pull requests.

Reply:
I enabled single precision (cmake .. -DG2O_SINGLE_PRECISION_MATH=YES) to assess what happens, and I get multiple build errors. These tend to be of the form:

/Users/ucacsjj/Proj/Other/g2o_master/g2o/g2o/apps/g2o_hierarchical/g2o_hierarchical_test_functions.cpp:81:17: error: no matching member function for call to 'computeMarginals'
  if (optimizer.computeMarginals(spinv, blockIndices)) {
      ~~~~~~~~~~^~~~~~~~~~~~~~~~
/Users/ucacsjj/Proj/Other/g2o_master/g2o/g2o/core/sparse_optimizer.h:131:10: note: candidate function not viable: no known conversion from 'SparseBlockMatrix<Eigen::MatrixXd>'
      (aka 'SparseBlockMatrix<Matrix<double, Dynamic, Dynamic> >') to 'SparseBlockMatrix<g2o::MatrixX> &' (aka 'SparseBlockMatrix<Matrix<float, Eigen::Dynamic, Eigen::Dynamic, Eigen::ColMajor> > &') for 1st
      argument
    bool computeMarginals(SparseBlockMatrix<MatrixX>& spinv, const std::vector<std::pair<int, int> >& blockIndices);

Have you experienced these?

Reply:
I'm working on another branch with that enabled and I haven't tested HEAD in a while so maybe someone could have broken it since then. If the cst function is reachable and builds in double precision, there's no reason for it not to when running single precision so I don't think it should block you. Although, you could just leave it as is and call it a day, it doesn't really matter.

Reply:
I did a bit more checking. (There are a very large number of files in which the floating point 0., 1., 1000., etc are assigned to number_t files.)

If I have the test code:

int main(int argc, const char* argv[])
{
  float f = 1.;  
}

I get no warnings (even using -Wconversion for both clang and gcc). However, if I change the value to be a floating point in any way, e.g.,

int main(int argc, const char* argv[])
{
  float f = 1.0001;
}

Then warnings do appear.

Therefore, it appears that warnings about casting will arise if the literal is a "non-integer" number. From what I can tell, all the outstanding literals in the code are of this integer form, and so using cst doesn't appear to be required.